src/rust/engine/options/src/lib.rs

```diff
@@ -108,6 +109,8 @@ pub(crate) trait OptionsSource: Send + Sync {
     ///
     fn display(&self, id: &OptionId) -> String;
 
+    fn as_any(&self) -> &dyn Any;
```

Comment:
This allows downcasting of the `dyn OptionSource` to a ConfigReader, when we know it is one.

Reply:
Potentially the trait could inherit from `Any`, so `dyn OptionSource` "is" a `dyn Any` too? 

Just polish though, no need to change.

Reply:
Having the trait inherit `Any` is not sufficient, since `downcast` is not a method of the Any trait, so it isn't implemented on `OptionSource` (It's a method on the trait object `dyn Any`). And Rust won't let you upcast to `dyn Any` (an `Arc<dyn OptionSource>` is not an `Arc<dyn Any>`). 

The `as_any()` idiom is the canonical way to get at the `Any` implementation of the underlying concrete struct.

Reply:
Ah, I see. Thanks